File: app/src/main/java/com/example/kocja/rabbiter_online/adapters/UpcomingEventsAdapter.java
package com.example.kocja.rabbiter_online.adapters;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.kocja.rabbiter_online.R;

import java.util.ArrayList;
import java.util.List;


public class UpcomingEventsAdapter extends RecyclerView.Adapter<UpcomingEventsAdapter.viewHolder> {
    private List<String> eventList;
    private onClickListen listener;
    private final boolean clickable;
    public UpcomingEventsAdapter(List<String> events,boolean clickable){
        eventList = new ArrayList<>(events.size());
        eventList = events;
        this.clickable = clickable;
    }

    class viewHolder extends RecyclerView.ViewHolder implements View.OnClickListener{
        final TextView eventString;
        viewHolder(View itemView) {
            super(itemView);
            eventString = itemView.findViewById(R.id.UpcomingText);
            eventString.setOnClickListener(this);
        }
         @Override
         public void onClick(View view) {
            if(clickable){
                listener.onItemClick(view,getAdapterPosition());
            }
         }
     }
    @NonNull
    @Override
    public viewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.upcoming_event_layout,parent,false);
        return new viewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull viewHolder holder, int position) {
        holder.eventString.setText(eventList.get(position));

    }

    @Override
    public int getItemCount() {
        return eventList.size();
    }

    public void setLongClickListener(onClickListen listen){
        listener = listen;
    }

    public interface onClickListen{
        void onItemClick(View view, int position);
    }

}

File: app/src/main/java/com/example/kocja/rabbiter_online/adapters/UpcomingEventsAdapter.kt
package com.example.kocja.rabbiter_online.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.example.kocja.rabbiter_online.databinding.UpcomingEventLayoutBinding


class UpcomingEventsAdapter(private val eventList: List<String>) : RecyclerView.Adapter<UpcomingEventsAdapter.ViewHolder>() {
    private var listener: View.OnClickListener? = null


    inner class ViewHolder(val item: UpcomingEventLayoutBinding) : RecyclerView.ViewHolder(item.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = UpcomingEventLayoutBinding.inflate(LayoutInflater.from(parent.context),parent,false)
        if(listener != null) {
            view.root.setOnClickListener(listener)
        }
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.item.UpcomingText.text = eventList[position]
    }

    override fun getItemCount(): Int {
        return eventList.size
    }

    fun setLongClickListener(listen: View.OnClickListener) {
        listener = listen
    }

}
